### ISSUE TITLE
Rust vsprintf wrapper off-by-one error

### ISSUE THREAD

**Issue by stevenengler** [shadow/shadow#1432]
I wrote a long bug report and then accidentally clicked on a link and lost it, so I'll keep this short :)

https://github.com/shadow/shadow/blob/d62d1b8adac65ea4d31e08cfb17116e34c9462e1/src/main/host/thread_ptrace.c#L717-L718

Causes a bug in the Rust vsprintf wrapper where it adds a nul character to the vector returned by `vsprintf_raw()`.

Example hex dump. Notice the nul character where there is supposed to be a `'` after the hostname for the above log message:

```text
00000e60  61 74 61 2f 68 6f 73 74  73 2f 34 75 74 68 6f 72  |ata/hosts/4uthor|
00000e70  69 74 79 32 00 0a 30 30  3a 30 30 3a 33 34 2e 32  |ity2..00:00:34.2|
```

I believe this is caused by:

https://github.com/sporksmith/vsprintf/blob/ad7621914b24bf8cfaa3b67abb9aa41c470ba6d1/src/lib.rs#L55-L58

Since the return value from `vsnprintf()` should include the nul character.

**Comment by stevenengler** [comment]
I assigned you @sporksmith since it's probably easiest for you to change.

**Comment by sporksmith** [comment]
Sorry for the subtle bug and thanks for tracking it down! I *thought* that if the buffer provided to vsnprintf wasn't big enough, it wouldn't NULL terminate. It appears I either misremembered or they removed that footgun at some point :)

Fixed in 1269b19

**Comment by stevenengler** [comment]
Thanks!
